src/frontend/packages/core/src/shared/components/list/data-sources-controllers/local-filtering-sorting.ts

```diff
@@ -30,11 +30,14 @@ function getFilterFunction(def: DataFunctionDefinition): DataFunction<any> {
     }
     return entities.filter(e => {
       e = extractActualListEntity(e);
-      const value = getValue(e, fieldArray, 0, true);
+      let value = getValue(e, fieldArray, 0, true);
       if (!value) {
         return false;
       }
-      return value.toUpperCase().includes(upperCaseFilter);
+      if (Array.isArray(value)) {
+        value = value.join(",")
+        return value.toUpperCase().includes(upperCaseFilter);
+      }
```

Comment:
```suggestion
      if (Array.isArray(value)) {
        value = value.join(",")
      }
      
      return value.toUpperCase().includes(upperCaseFilter);
```
the return still needs to happen even if its not a list, otherwise this will break the filtering

the tests dont seem to cover this. could you please add another test where you check that the text filtering still works without tags?